tests/docker/Dockerfile

```diff
@@ -64,6 +64,16 @@ RUN mkdir -p "/opt/kafka-2.3.1" && chmod a+rw /opt/kafka-2.3.1 && curl -s "$KAFK
     make install && \
     ldconfig
 
+# install dependencies for kafka clients such as sarama
+RUN go get github.com/Shopify/sarama && \
```

Comment:
I'm using `go get` instead of `go install` because go had an issue with the syntax necessary to execute go install.

Reply:
@twmb mind taking a quick peek at this?

Reply:
So after looking:
- Apt lists are often out of date, the recommended way to install Go is to just download the latest version from the dl.golang.org page
- This apt is using Go 1.14 currently, which has less support for `go install` in module aware mode
- Go 1.16 would allow the install @latest syntax to work
- Go 1.14 is officially not maintained anymore, so we should bump to the latest now
- I sent @NyaliaLui a few lines of makefile code that could be cobbled together to always use the lates Go version in this dockerup
- We need to ping @ivotron to see what he thinks on this

For context, the shell code to do this is roughly:

```make
GOLANG_LATEST=$(curl -sL https://golang.org/dl/?mode=json | jq 'map(select (.stable == true))[0].files[] | select (.os == "linux" and .arch == "amd64")'
VERSION=$(echo "$$GOLANG_LATEST" | jq -r '.version');
SHA256=$(echo "$$GOLANG_LATEST" | jq -r '.sha256');
echo "${GOLANG_SHA256}  ${GOLANG_VERSION}.linux-amd64.tar.gz" | sha256sum -c 
curl -fsSLO https://storage.googleapis.com/golang/${VERSION}.linux-amd64.tar.gz
```

cc @ivotron

Reply:
agreed. Another thing that can be done is to add a `COPY ./taskfiles/dev.yml` to the Dockerfile, and run `task dev:install-go` when the image is built. In this way we don't need to have duplicated logic to do the same thing (installing golang in this case).

I'm happy to help with this @NyaliaLui

Reply:
I'll put the change in no problem.